File: Documentacao/LIA2_008_PedagogicalAdaptive_Mobile_Thinking_FIX02/apps/studentWeb/src/contracts/materialContract.ts
export type MaterialStatus =
  | 'UPLOADED'
  | 'PROCESSING'
  | 'PARTIAL'
  | 'READY'
  | 'ERROR'
  | 'ARCHIVED'

export interface MaterialContract {
  contractName: 'Material.v3'
  materialId: string
  studentId: string
  studentLearningContextId: string | null
  studentSubjectId: string | null
  studentLearningUnitId: string | null
  title: string
  materialType: 'PDF' | 'IMAGE' | 'TEXT' | 'DOCUMENT' | 'OTHER'
  sourceType: 'UPLOAD' | 'MANUAL' | 'LINK'
  description: string | null
  status: MaterialStatus
  analysisRequested: boolean
  studyEnabled: boolean
  requestedModelId: string | null
  aiMode: 'AUTO' | 'FIXED' | 'CUSTOM'
  fixedModelId: string | null
  textModelId: string | null
  visionModelId: string | null
  embeddingModelId: string | null
  thinkingMode: 'AUTO' | 'ON' | 'OFF'
  sourceGroupId: string | null
  sourceSequence: number | null
  lastProcessingErrorCode: string | null
  lastProcessingErrorMessage: string | null
  createdAt: string
  updatedAt: string
}

export interface MaterialUploadResultContract {
  contractName: 'MaterialUploadResult.v2'
  material: MaterialContract
  file: {
    contractName: 'MaterialFile.v1'
    materialFileId: string
    materialId: string
    originalFileName: string
    mimeType: string
    sizeBytes: number
    sha256: string
    status: 'ACTIVE' | 'SUPERSEDED' | 'ERROR'
    createdAt: string
  }
  documentId: string | null
  documentVersionId: string | null
  pageCount: number
  textBlockCount: number
  visualPendingCount: number
  chunkCount: number
  analysisPerformed: boolean
}

export interface MaterialBatchItemContract {
  contractName: 'MaterialBatchItem.v1'
  fileName: string
  success: boolean
  result: MaterialUploadResultContract | null
  errorCode: string | null
  errorMessage: string | null
}

export interface MaterialBatchUploadResultContract {
  contractName: 'MaterialBatchUploadResult.v1'
  totalFiles: number
  successCount: number
  errorCount: number
  items: MaterialBatchItemContract[]
}

File: apps/backend/app/contracts/materialContract.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class MaterialContract(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    contractName: Literal["Material.v3"] = "Material.v3"
    materialId: UUID
    studentId: UUID
    studentLearningContextId: UUID | None
    studentSubjectId: UUID | None
    studentLearningUnitId: UUID | None
    title: str
    materialType: Literal["PDF","IMAGE","TEXT","DOCUMENT","OTHER"]
    sourceType: Literal["UPLOAD","MANUAL","LINK"]
    description: str | None
    status: Literal["UPLOADED","PROCESSING","PARTIAL","READY","ERROR","ARCHIVED"]
    analysisRequested: bool
    studyEnabled: bool
    sourceFileRetained: bool = True
    requestedModelId: str | None
    aiMode: Literal["AUTO","FIXED","CUSTOM"] = "AUTO"
    fixedModelId: str | None = None
    textModelId: str | None = None
    visionModelId: str | None = None
    embeddingModelId: str | None = None
    thinkingMode: Literal["AUTO","ON","OFF"] = "AUTO"
    sourceGroupId: UUID | None = None
    sourceSequence: int | None = None
    lastProcessingErrorCode: str | None
    lastProcessingErrorMessage: str | None
    createdAt: datetime
    updatedAt: datetime


class MaterialFileContract(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    contractName: Literal["MaterialFile.v1"] = "MaterialFile.v1"
    materialFileId: UUID
    materialId: UUID
    originalFileName: str
    mimeType: str
    sizeBytes: int
    sha256: str
    status: Literal["ACTIVE","SUPERSEDED","ERROR"]
    createdAt: datetime


class MaterialUploadResultContract(BaseModel):
    contractName: Literal["MaterialUploadResult.v2"] = "MaterialUploadResult.v2"
    material: MaterialContract
    file: MaterialFileContract
    documentId: UUID | None
    documentVersionId: UUID | None
    pageCount: int
    textBlockCount: int
    visualPendingCount: int
    chunkCount: int
    analysisPerformed: bool


class MaterialBatchItemContract(BaseModel):
    contractName: Literal["MaterialBatchItem.v1"] = "MaterialBatchItem.v1"
    fileName: str
    success: bool
    result: MaterialUploadResultContract | None = None
    errorCode: str | None = None
    errorMessage: str | None = None


class MaterialBatchUploadResultContract(BaseModel):
    contractName: Literal[
        "MaterialBatchUploadResult.v1"
    ] = "MaterialBatchUploadResult.v1"
    totalFiles: int
    successCount: int
    errorCount: int
    items: list[MaterialBatchItemContract]


class MaterialStudyUsageUpdateContract(BaseModel):
    contractName: Literal[
        "MaterialStudyUsageUpdate.v1"
    ] = "MaterialStudyUsageUpdate.v1"
    studyEnabled: bool



class MaterialModelPreferenceUpdateContract(BaseModel):
    contractName: Literal[
        "MaterialModelPreferenceUpdate.v1"
    ] = "MaterialModelPreferenceUpdate.v1"
    requestedModelId: str | None = None
